Reorganize tensor mechanics documentation

sapitts opened the pull request:
The landing page for the tensor mechanics module documentation was poorly organized with images that did not work nicely around the text, and this commit reorganizes the images to a single row on the initial tensor mechanics documentation page. The submodule update is for a new version of the rpv example image.

Additionally this commit moves some documentation for two actions into the *Action/index.md file. @bwspenc and @aeslaughter please take a look at this newer organization and let me know what can be improved with how and where we include documentation for the actions.

Refs #10516

sapitts committed:
Reformat tensor mechanics landing page, some actions, Refs #10516

sapitts committed:
large_media submodule update, Refs #10516

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit eb7cc99c84687a89ee99287d8a9db7cae14ef28c
+Subproject commit 52bb06c5f75001d52fc7746a9df0274567079637
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/190775/) on e6fe314 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11522/site)

This comment will be updated on new commits.

bwspenc commented on modules/tensor_mechanics/doc/content/documentation/modules/tensor_mechanics/index.md:
```diff
@@ -1,25 +1,19 @@
 - Tensor implementation matches mathematical theory
 - Straight-forward procedure for adding new physics
-- Complete [System Reference List](tensor_mechanics/systems.md) for the tensor mechanics code
+- Complete [Module Documentation List](tensor_mechanics/systems.md) for the tensor mechanics code
```
Maybe this line isn't needed since there's a link to the same thing up above now?

bwspenc commented on modules/tensor_mechanics/doc/content/documentation/systems/BCs/Pressure/index.md:
```diff
@@ -1,6 +1,40 @@
 # Pressure System
 
-The Pressure Action is used to create a set of pressure boundary conditions for a string of displacement variables; the typical use case for this action is the application of hydrostatic pressure.
+!syntax description /BCs/Pressure/PressureAction
```
I think I like the way you have it set up better now. I'm curious how you like it. One thing that's weird is that the path to the Pressure system page and the regular Pressure BC looks identical in the breadcrumb list at the top of the page.

sapitts commented:
I've pushed this commit again, but I'd like to move all of the action documentation into the */index.md files for all the documented tensor mechanics actions in this one pull request if this is the direction we'd like to go. 

@dschwen, @novasr is this shuffling of the main information for actions into the */index.md file for each action okay with you?

sapitts commented:
@sveerara and @hchen139 I've moved around and reformatted (into a table) the documentation for LineElementAction and GeneralizedPlaneStrain. Would you take a look at this commit and make sure the documentation is still correct please?

sveerara commented on modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/LineElementMaster/index.md:
```diff
@@ -1 +1,73 @@
+| Calculation of stress divergence for a +Truss+ element | [StressDivergenceTensorsTruss](/StressDivergenceTensorsTruss.md) | `displacements` : a string of the displacement variables |
+
+This action can also optionally set up the AuxVariables and AuxKernels combinations for post processing outputs. See the documentation for the [TensorMechanics Master action](../Master) for the generation of output AuxVariables and AuxKernels.
+
```
The Line element action does not set up aux variables and aux kernels for post-processing output yet. Also, there is no stress/strain tensor for beam and truss elements. So, I don't think the options in Tensor Mechanics Master Action can be used to set up the required object for post processing.

sapitts replied:
Thank you for catching this mistake @sveerara! I'll get it fixed in my next push

sveerara commented on modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/LineElementMaster/index.md:
```diff
@@ -1 +1,73 @@
+
+
+This action can also optionally set up the AuxVariables and AuxKernels combinations for post processing outputs. See the documentation for the [TensorMechanics Master action](../Master) for the generation of output AuxVariables and AuxKernels.
+
```
Same comment as above.

sveerara reviewed (changes requested):
@sapitts, thanks for reorganizing the line element action documentation. This looks really good. I have a couple of minor comments regarding the line element action.

hchen139 commented on modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/index.md:
```diff
@@ -1,38 +1,44 @@
+
+## Constructed MooseObjects
+
+The `GeneralizedPlaneStrain` Action is used to construct the kernels, strain materials, displacement variables, and user objects in a consistent manner as required for a generalized plane strain simulation.
```
This action doesn't setup the strain materials, displacement variables

sapitts replied:
Action documentation is a bit tricky because we hid so much of the normal inheritance, ect. from the user. I had in mind writing the action documentation as a 'one-stop shop' for the reader/user: I wanted to put all of the information that a user would need to completely use the GeneralizedPlaneStrain action on this single page. This way, the user won't need to go to the tensor mechanics master action page to find out which parameters to add for displacements and strain formulations. As I'm looking at this again, I realize that I can't then cheat with sending the user to tensor mechanics master action for auxkernels and auxvariables either--this documentation information should be on the gps page if I include the displacements and strains information.

@bwspenc @novasr How did each of you envision the documentation for the gps action? Do you see this documentation as a self contained 'here's everything you need to know about the gps action'? Or would you prefer that the gps action page has just what the gps adds onto the master action page and that we send users to the master action page for information about adding variables, strain formulations, and auxvariables/auxkernels? I'll address @hchen139's comments based on if inclusive or specialized gps action make more sense

Thanks for bringing up this point @hchen139!

hchen139 commented on modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/index.md:
```diff
@@ -1,38 +1,44 @@
+!table id=generalizedPS_action_table caption=Correspondence Among Action Functionality and MooseObjects for the `GeneralizedPlaneStrain` Action
+| Functionality     | Replaced Classes   | Associated Parameters   |
+|-------------------|--------------------|-------------------------|
+| Scalar out-of-plan strain coupling | [Generalized Plane Strain Off-diagonal Kernel](/GeneralizedPlaneStrainOffDiag.md) | `displacements` : a string of the displacement variables |
```
Functionality: In-plane (displacements or temperature) and out-of-plane (scalar strain) coupling
Associated Parameters: `displacements `: a string of the displacement variables; `temperature`: a string of the temperature variables; `scalar_out_of_plane_strain`: a string of the scalar out-of-plane strain variable

sapitts commented:
@hchen139 and @bwspenc I've updated the documentation for gps to include the corrections to the coupling that @hchen139 gave me, and I've added in more documentation for the capabilities that the gps action inherits from the tensor mechanics master action (adding variables, strain calculators, and generate output). I think this documentation of inherited capabilities follows what we've done with other models, e.g. creep in bison, and would best address @novasr's point about being able to trace the use of an action in the input file to the PDE equations.  Would you guys give this another read through and let me know what you like and dislike please?

hchen139 commented on modules/tensor_mechanics/doc/content/documentation/systems/Modules/TensorMechanics/GeneralizedPlaneStrain/index.md:
```diff
@@ -1,42 +1,54 @@
+!table id=generalizedPS_action_table caption=Correspondence Among Action Functionality and MooseObjects for the `GeneralizedPlaneStrain` Action
+| Functionality     | Replaced Classes   | Associated Parameters   |
+|-------------------|--------------------|-------------------------|
+| In-plane equilibrium conditions for the given coordinate system | [StressDivergenceTensors](/Kernels/StressDivergenceTensors.md) or [StressDivergenceRZTensors](/Kernels/StressDivergenceRZTensors.md) or [StressDivergenceRSphericalTensors](/Kernels/StressDivergenceRSphericalTensors.md) | `displacements` : a string of the displacement field variables |
```
Do the generalized plane strain assumptions apply to an spherically symmetric problem? I think we only have that for the XY and RZ problems.

sapitts commented:
@hchen139 thank you for all of your comments and corrections.  I think this commit should finally have everything correct (I confused the gps action with the layered1d action in bison), and I'm sorry I didn't follow your comments the first time.

sapitts commented:
@bwspenc this is ready for review as you have time-- thanks!

sapitts commented:
Thank you @bwspenc!
